[3.2] docs(colorpallete): Revamped ColorPallete Overview (#826)
telerik/blazor-docs: components/colorpalette/overview.md

>caption Basic color palette with two-way value binding and a default predefined palette.

````CSHTML
<TelerikColorPalette @bind-Value="@MyColor" />

<p>Selected color: <span style="color: @MyColor">@MyColor</span></p>

@code {
    string MyColor { get; set; }
}
````

## Predefined Colors

The ColorPalette component comes with a set of predefined colors that are shown to the users. [Read more about the available ColorPallete predefined colors]({%slug colorpalette-presets%}).

## Custom Colors

The ColorPallete can work with your own set of colors. [Read more about the Blazor ColorPallete custom colors setup]({%slug colorpalette-custom-colors%}).

## Events

The Blazor ColorPalette fires value change and blur events that you can handle and further customize its behavior. [Read more about the Blazor Dialog events]({%slug colorpalette-events%}).

## ColorPalette Parameters

The Blazor ColorPalette provides various parameters to configure the component:

<style>
    article style + table {
        table-layout: auto;
        word-break: normal;
    }
</style>
| Parameter | Type and Default Value | Description |

> | ----------- | ----------- | ----------- |

Reviewer: Either format the table via ASCII art, or use minimal design. No need for something in between.
```suggestion
| --- | --- | --- |
```

}
````

## Predefined Colors

> The ColorPalette component comes with a set of predefined colors that are shown to the users. [Read more about the available ColorPallete predefined colors]({%slug colorpalette-presets%}).

Reviewer: ```suggestion
The ColorPalette component comes with multiple sets of predefined colors that are shown to the users. [Read more about the available predefined ColorPallete colors]({%slug colorpalette-presets%}).
```

## Custom Colors

The ColorPallete can work with your own set of colors. [Read more about the Blazor ColorPallete custom colors setup]({%slug colorpalette-custom-colors%}).

## Events

The Blazor ColorPalette fires value change and blur events that you can handle and further customize its behavior. [Read more about the Blazor Dialog events]({%slug colorpalette-events%}).

## ColorPalette Parameters

> The Blazor ColorPalette provides various parameters to configure the component:

Reviewer: ```suggestion
The Blazor ColorPalette provides various parameters to configure the component. Also check the [ColorPalette public API](/blazor-ui/api/Telerik.Blazor.Components.TelerikColorPalette).
```

<style>
    article style + table {
        table-layout: auto;
        word-break: normal;
    }
</style>
| Parameter | Type and Default Value | Description |
| ----------- | ----------- | ----------- |
| `Class` | `string` | Renders a custom CSS class to the `<div class="k-colorpalette">` element. |

> | `Colors` | `IEnumerable<string>` <br /> (`Office`) | The collection of colors the user can choose from. Can be one of the [presets that come with the component]({%slug colorpalette-presets%}), or [a custom list]({%slug colorpalette-custom-colors%}). |

Reviewer: ```suggestion
| `Colors` | `IEnumerable<string>` <br /> (`Office`) | The collection of colors for the user to choose from. Can be one of the [built-in preset palettes]({%slug colorpalette-presets%}), or [a custom list of colors]({%slug colorpalette-custom-colors%}). |
```

| `Columns` | `int` <br /> (`10`) | The number of columns to use when rendering the Colors list. Determines the size of the component together with the `TileHeight` and `TileWidth`. |

> | `Enabled` | `bool` | Whether the component is enabled. |

Reviewer: The default boolean type value is false.

```suggestion
| `Enabled` | `bool` <br /> (`true`) | Whether the component is enabled. |
```

> | `Id` | `string` | Renders as the `id` attribute to the `<div class="k-colorpalette">` element. |

Reviewer: ```suggestion
| `Id` | `string` | Renders as an `id` attribute of the `<div class="k-colorpalette">` element. |
```

> | `TabIndex` | `Nullable<int>` | Maps to the `tabindex` attribute to the `<div class="k-colorpalette">` element. You can use it to customize the order in which the elements in your page focus with the `Tab` key. |

Reviewer: ```suggestion
| `TabIndex` | `Nullable<int>` | Maps to the `tabindex` attribute of the `<div class="k-colorpalette">` element. Use it to customize the tabbing focus order on the page. |
```

> | `TileHeight` | `string` | The height of each individual color item. Determines the size of the component together with the `Columns` and `TileWidth`. Can take CSS [dimensions]({%slug common-features/dimensions%}) strings  |

Reviewer: ```suggestion
| `TileHeight` | `string` | The height of each individual color item. Determines the size of the component together with the `Columns` and `TileWidth`. Accepts [CSS dimensions]({%slug common-features/dimensions%}) strings.  |
```

> | `TileWidth` | `string` | The width of each individual color item. Determines the size of the component together with the `Columns` and `TileHeight`. Can take CSS [dimensions]({%slug common-features/dimensions%}) strings  |

Reviewer: ```suggestion
| `TileWidth` | `string` | The width of each individual color item. Determines the size of the component together with the `Columns` and `TileHeight`. Accepts [CSS dimension]({%slug common-features/dimensions%}) strings. |
```

> | `Value` | `string` | Get/set the value of the input, can be used for binding. Can take any string that can be a [CSS background-color string](https://css-tricks.com/almanac/properties/b/background-color/). The presets we provide use hex format (`#123abc`). |

Reviewer: ```suggestion
| `Value` | `string` | Sets the value of the input, can be used for binding. Accepts any valid [CSS `background-color` string](https://css-tricks.com/almanac/properties/b/background-color/). The preset palettes use HEX format (`#123abc`). |
```
